### PR TITLE
engine: remove MK dependency

### PR DESCRIPTION
**Pull request by bassosimone**
Here's another experiment where we're using the API at https://github.com/ooni/probe-engine/issues/893#issuecomment-689031449. The experiment is not complete, but it is possible to clearly see that we are easily and completely able to create wrappers for the Go code such that the Java code only sees interfaces. This is better than the previous experiment because we can really decouple Java code and Go code, which means we can write lots of tests.

Closes https://github.com/ooni/probe-engine/issues/923.

### CODE REVIEW

File: engine/src/main/java/org/openobservatory/engine/PESession.java [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+    public OONIGeolocateTask newGeolocateTask(long timeout) throws OONIException {
+        try {
+            return new PEGeolocateTask(session.newGeolocateTask(timeout));
+        } catch (Exception exc) {
+            throw new OONIException("session.newGeolocateTask failed", exc);
+        }
+    }
+
+    @Override
+    public OONIProbeServicesClient newMakeSubmitterTask(long timeout) throws OONIException {
```
Comment (bassosimone) [inline engine/src/main/java/org/openobservatory/engine/PESession.java:27]: This should be `newProbeServicesClient`

File: app/src/main/java/org/openobservatory/ooniprobe/common/ResubmitTask.java [hunk 2]
```diff
@@ -39,38 +40,29 @@
...
-            m.is_upload_failed = false;
-            m.save();
-        } else {
-            Log.w(CollectorTask.class.getSimpleName(), results.getLogs());
+        try {
+            input = FileUtils.readFileToString(file, Charset.forName("UTF-8"));
+            try (OONISubmitMeasurementTask task = submitter.newSubmitMeasurementTask(uploadTimeout)){
+                OONISubmitResults results = task.run(input);
+                String output = results.updatedMeasurement;
+                FileUtils.writeStringToFile(file, output, Charset.forName("UTF-8"));
```
Comment (bassosimone) [inline app/src/main/java/org/openobservatory/ooniprobe/common/ResubmitTask.java:54]: We can simplify the code by not using `output` here

File: app/src/main/java/org/openobservatory/ooniprobe/common/ResubmitTask.java [hunk 3]
```diff
@@ -107,23 +99,27 @@ protected Boolean doInBackground(Integer... params) {
...
-                break;
-            String paramOfParam = activity.getString(R.string.paramOfParam, Integer.toString(i + 1), Integer.toString(measurements.size()));
-            publishProgress(activity.getString(R.string.Modal_ResultsNotUploaded_Uploading, paramOfParam));
-            Measurement m = measurements.get(i);
-            m.result.load();
-            try {
-                if (!perform(activity, m)){
+        try (OONISession session = Engine.newSession(Engine.getDefaultSessionConfig(getActivity(), BuildConfig.SOFTWARE_NAME, BuildConfig.VERSION_NAME, logger));
+             OONIProbeServicesClient client = session.newMakeSubmitterTask(30);
+             OONIReport submitter = client.run();){
```
Comment (bassosimone) [inline app/src/main/java/org/openobservatory/ooniprobe/common/ResubmitTask.java:104]: The `run` method of `OONIProbeServicesClient` needs to become `openReport`

File: app/src/main/java/org/openobservatory/ooniprobe/test/TestAsyncTask.java [hunk 4]
```diff
@@ -90,6 +89,7 @@ protected TestAsyncTask(ACT activity, Result result) {
 				}
 			} catch (Exception e) {
 				publishProgress(ERR, act.getString(R.string.Modal_Error_CantDownloadURLs));
+				e.printStackTrace();
```
Comment (bassosimone) [inline app/src/main/java/org/openobservatory/ooniprobe/test/TestAsyncTask.java:92]: Shouldn't we pass the exception to the exception manager here?

File: engine/src/main/java/org/openobservatory/engine/Engine.java [hunk 5]
```diff
@@ -2,81 +2,81 @@
...
+        } catch (Exception exc) {
+            throw new OONIException("cannot start OONI Probe Engine task", exc);
         }
-        return new MKExperimentTaskAdapter(io.ooni.mk.MKAsyncTask.start(settings.serialization()));
     }
 
-    /** maybeUpdateResources updates the bundled resources if needed */
-    public static boolean maybeUpdateResources(Context context) {
-        return io.ooni.mk.MKResourcesManager.maybeUpdateResources(context);
+    /** resolveProbeCC creates a new GeoIP lookup task, runs it and return the probeCC. */
```
Comment (bassosimone) [inline engine/src/main/java/org/openobservatory/engine/Engine.java:28]: `creates` => `returns`

**Review by bassosimone — COMMENTED** [review]
I suggested small refactoring actions aimed at improving consistency.

File: engine/build.gradle [hunk 6]
```diff
@@ -20,13 +20,16 @@ android {
             proguardFiles getDefaultProguardFile('proguard-android-optimize.txt'), 'proguard-rules.pro'
         }
     }
+    compileOptions {
+        sourceCompatibility JavaVersion.VERSION_1_8
+        targetCompatibility JavaVersion.VERSION_1_8
+    }
```
[the remark this answers is not in the document]
Comment (lorenzoPrimi) [reply engine/build.gradle:26]: This was added by you, don't ask me why :D

File: engine/src/main/java/org/openobservatory/engine/ArrayLogger.java [hunk 7]
```diff
@@ -0,0 +1,29 @@
+package org.openobservatory.engine;
+
+import java.util.ArrayList;
+
+public class ArrayLogger extends AndroidLogger {
```
[the remark this answers is not in the document]
Comment (lorenzoPrimi) [reply engine/src/main/java/org/openobservatory/engine/ArrayLogger.java:5]: Actually I don't care about `Log.d` at all and neither the user as it's a debug flag onlyI just thought was a placeholder class you put there to "make it do something" later

File: engine/src/main/java/org/openobservatory/engine/Engine.java (hunk 5 above)
Comment (lorenzoPrimi) [reply engine/src/main/java/org/openobservatory/engine/Engine.java:28]: read the entire description

File: engine/build.gradle (hunk 6 above)
[the remark this answers is not in the document]
Comment (bassosimone) [reply engine/build.gradle:26]: I guess this was a mistake or an experiment

File: engine/src/main/java/org/openobservatory/engine/Engine.java (hunk 5 above)
Comment (bassosimone) [reply engine/src/main/java/org/openobservatory/engine/Engine.java]: lol - still I think the documentation should not mention implementation details and only say what this function does

**Review by bassosimone — COMMENTED** [review]
okay, I guess this is good to go now

File: engine/build.gradle (hunk 6 above)
[the remark this answers is not in the document]
Comment (lorenzoPrimi) [reply engine/build.gradle:26]: btw we have the same code in the app module

[the remark this answers is not in the document]
Comment (bassosimone) [reply engine/build.gradle:26]: ah, lol, then it's probably fine to keep it!

**Pull request merged**